### PR TITLE
refactor alerts and move to storybook

### PR DESCRIPTION
**Pull request by rrhyne**
Refactors PLG alerts to reduce complexity, remove code, and make them more modular and specific in terms of how styles are applied. For instance, titles are now passed through property instead of children, which reduces the need for devs to research what header the alert should be. 

Additionally, this PR creates a storybook file to show various alert content to allow for easier manipulation and review. 

## Test plan

Manually ensure all alerts display as they should.

### CODE REVIEW

File: client/web/src/cody/components/CodyAlert.module.scss [hunk 1]
```diff
@@ -1,117 +1,63 @@
...
-    /* stylelint-disable-next-line declaration-property-unit-allowed-list */
-    padding-left: calc(1.5rem + 135px + 1.5rem);
-    padding-bottom: 1rem;
+.purple {
     background-color: var(--violet-01);
-    color: var(--gray-11);
-    overflow: hidden;
+    color: var(--violet-06);
+
+    border:1px solid #D089EC;
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.module.scss:37]: Do we have a corresponding variable in [colors.scss](https://github.com/sourcegraph/sourcegraph/blob/ba9e78eae42650fd990c3810d5caa93a94a69dea/client/wildcard/src/global-styles/colors.scss)?

File: client/web/src/cody/components/CodyAlert.module.scss [hunk 2]
```diff
@@ -1,117 +1,63 @@
...
 }
 
 .error {
-    /* stylelint-disable-next-line declaration-property-unit-allowed-list */
-    padding-left: calc(2rem + 124px + 2rem);
-    padding-bottom: 1rem;
     background-color: var(--oc-orange-1);
     color: var(--gray-08);
-    overflow: hidden;
+    border:1px solid #FFA94D;
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.module.scss:58]: nit: [`var(--border-width)`](https://github.com/sourcegraph/sourcegraph/blob/9e63554e5f679cc5520c4ee43f188473c2e011a3/client/wildcard/src/global-styles/variables.scss#L19) should work too.

File: client/web/src/cody/components/CodyAlert.tsx [hunk 3]
```diff
@@ -2,24 +2,31 @@ import React from 'react'
...
 import classNames from 'classnames'
 
+import { AlertBadge } from './AlertBadge'
+import { CodyProBadge } from './CodyProBadge'
+
 import styles from './CodyAlert.module.scss'
 
 interface CodyAlertProps extends React.HTMLAttributes<HTMLDivElement> {
-    variant: 'purple' | 'greenSuccess' | 'purpleSuccess' | 'purpleCodyPro' | 'greenCodyPro' | 'error'
+    variant: 'purple' | 'green' | 'error'
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.tsx:11]: Thanks for removing non-descriptive variants!

Are purple and green intended for different use cases (e.g., green for success and purple for info)? If yes, we can use these statuses as variant values: `variant: 'success' | 'info' | 'error'`. In this way, we'll be more consistent with [`Alert` variants](https://github.com/sourcegraph/sourcegraph/blob/9e63554e5f679cc5520c4ee43f188473c2e011a3/client/wildcard/src/components/Alert/constants.ts#L1-L11).

File: client/web/src/cody/components/CodyAlert.module.scss [hunk 4]
```diff
@@ -1,117 +1,63 @@
...
 
-    &::after {
-        content: '';
-        position: absolute;
-    }
+.card {
+    height: 100%;
+    margin-top:auto;
+    margin-right:1rem;
 }
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.module.scss:13]: nit: can be replaced with utility classes `h-100 mt-auto mr-3` (I see you already use utility classes in other places of this PR)

File: client/web/src/cody/components/CodyAlert.module.scss [hunk 5]
```diff
@@ -1,117 +1,63 @@
...
-    :global(.theme-dark) & {
-        background-color: var(--pink);
-        color: var(--gray-08);
-    }
-    &::after {
-        // Purple checkmark SVG
-        background-image: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 31 25'><path fill='%23a405e1' d='M27 .4a3.4 3.4 0 0 0-2.4 1.1L12 15.6 7 9.3a3.4 3.4 0 0 0-5-.6 3.4 3.4 0 0 0-.5 4.9L9.2 23a3.4 3.4 0 0 0 5.3.1l15.2-17a3.4 3.4 0 0 0-.3-4.9 3.4 3.4 0 0 0-2.5-.8Z'/></svg>");
-    }
+.action-button-wrapper {
+    padding:1em;
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.module.scss:22]: I couldn't find this class usage. Do we need it?
Do we use `em` intentionally here?

File: client/web/src/cody/components/CodyAlert.tsx [hunk 6]
```diff
@@ -2,24 +2,31 @@ import React from 'react'
...
     className?: string
+    displayCard?: 'CodyPro' | 'Alert' | ''
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.tsx:13]:
``` suggestion
    badge?: 'pro' | 'alert'
```

File: client/web/src/cody/components/CodyAlert.tsx [hunk 7]
```diff
@@ -28,7 +35,14 @@ export const CodyAlert: React.FunctionComponent<CodyAlertProps> = ({ variant, cl
     return (
         // eslint-disable-next-line no-restricted-syntax
         <div className={alertClassName} {...props}>
-            {children}
+            {displayCard !== '' && (
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.tsx:38]:
```suggestion
            {displayCard && (
```

File: client/web/src/cody/components/CodyAlert.tsx [hunk 8]
```diff
@@ -2,24 +2,31 @@ import React from 'react'
...
     children: React.ReactNode
 }
 
-export const CodyAlert: React.FunctionComponent<CodyAlertProps> = ({ variant, className, children, ...props }) => {
+export const CodyAlert: React.FunctionComponent<CodyAlertProps> = ({
+    variant,
+    className,
+    title,
+    children,
+    displayCard = '',
```
Comment (taras-yemets) [inline client/web/src/cody/components/CodyAlert.tsx:22]:
```suggestion
    displayCard,
```

File: client/web/src/cody/invites/AcceptInviteBanner.tsx [hunk 9]
```diff
@@ -43,10 +43,8 @@ const AcceptInviteBannerContent: React.FC<{
     ) {
         return (
             <CodyAlert variant="error">
-                <H1 as="p" className="mb-2">
-                    Issue with invite
-                </H1>
-                <Text className="mb-0">The invitation is no longer valid. Contact your team admin.</Text>
+                <H1>Issue with invite</H1>
```
Comment (taras-yemets) [inline client/web/src/cody/invites/AcceptInviteBanner.tsx:46]: Do we intentionally remove `as="p"`? Do we want "Issue with invite" to be `<h1>`?

File: client/web/src/cody/management/subscription/new/CodyProCheckoutForm.tsx [hunk 10]
```diff
@@ -283,8 +282,8 @@ export const CodyProCheckoutForm: React.FunctionComponent<CodyProCheckoutFormPro
         <>
             {initialSeatCount + planChange.seatCountDiff >= 30 && (
                 <CodyAlert variant="purple">
-                    <H3>Explore an enterprise plan</H3>
-                    <Text className="mb-0">
+                    <H2>Explore an enterprise plan</H2>
```
Comment (taras-yemets) [inline client/web/src/cody/management/subscription/new/CodyProCheckoutForm.tsx:285]: Do we intentionally allow different heading tags and styles for `CodyAlert` heading?

File: client/web/src/cody/components/CodyAlert.tsx (hunk 3 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.tsx:11]: Currently they are used less semantically than that. Typically green is success, but I also use purple for success in upgrades to carry through from the purple upgrade CTA and differentiate from the green invite success. Let's leave it this way for now.

File: client/web/src/cody/components/CodyAlert.tsx (hunk 8 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.tsx:22]: Thanks... figuring out React again after a long break. :D

File: client/web/src/cody/components/CodyAlert.tsx (hunk 7 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.tsx:38]: I do find easier to read even if it isn't shorter... I don't have to determine if it's a boolean or just empty. I guess the IDE popover tell you that if you need to know though?

Comment (taras-yemets) [reply client/web/src/cody/components/CodyAlert.tsx:38]: TypeScript check will fail if `displayCard` is anything other than "CodyPro", "Alert" or `undefined`, see type definition a few lines higher (https://github.com/sourcegraph/sourcegraph/pull/63532#discussion_r1658466338)
```
displayCard?: 'CodyPro' | 'Alert' | ''
```

Comment (taras-yemets) [reply client/web/src/cody/components/CodyAlert.tsx:38]: I intentionally didn't mentioned the empty string, as we use it as a default empty value, and with the suggested change it's redundant IMO.

Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.tsx:38]: Ah, roger.

File: client/web/src/cody/components/CodyAlert.tsx (hunk 6 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.tsx:13]: Removing the empty string, but leaving the names the same to be 1:1 to the SVG filenames if that's ok with you.

File: client/web/src/cody/components/CodyAlert.module.scss (hunk 4 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.module.scss:13]: Done. Thanks!

File: client/web/src/cody/invites/AcceptInviteBanner.tsx (hunk 9 above)
Comment (rrhyne) [reply client/web/src/cody/invites/AcceptInviteBanner.tsx:46]: Fixing as per our convo by moving title into a dedicated prop and re-adding there.

File: client/web/src/cody/components/CodyAlert.module.scss (hunk 1 above)
Comment (rrhyne) [reply client/web/src/cody/components/CodyAlert.module.scss]: I've addressed as much of this as I can. Pushing for now!

**Review by taras-yemets — APPROVED** [review]

File: client/web/src/cody/management/CodyManagementPage.tsx [hunk 11]
```diff
@@ -94,73 +94,7 @@ export const CodyManagementPage: React.FunctionComponent<CodyManagementPageProps
         return null
     }
 
-    // Generate invite-related widgets for the page.
-    // The returned widgets depend on the user's role and subscription status.
-    // Only applicable when embedded Cody Pro UI is enabled.
-    const inviteWidgets = ((): {
```
Comment (taras-yemets) [inline client/web/src/cody/management/CodyManagementPage.tsx:100]: We still need this block.

File: client/web/src/cody/management/CodyManagementPage.tsx [hunk 12]
```diff
@@ -213,8 +192,7 @@ export const CodyManagementPage: React.FunctionComponent<CodyManagementPageProps
             <div className={classNames('border bg-1 mb-2', styles.container)}>
                 <CodyEditorsAndClients telemetryRecorder={telemetryRecorder} />
             </div>
-
-            <div className="pb-3" />
-        </Page>
-    )
+        </div>b
```
Comment (taras-yemets) [inline client/web/src/cody/management/CodyManagementPage.tsx:195]: `b` is redundant here. I think it's a merge leftover.

Comment (rrhyne) [reply client/web/src/cody/management/CodyManagementPage.tsx:195]: Fixed, TY.

**Comment by chrsmith** [comment]
Hey @rrhyne , @taras-yemets and I will take care of getting this PR cleaned up and merged.

**Comment by chrsmith** [comment]
Since we've stopped working on the newer Teams and Invites UI, perhaps we should just close this PR out.

**Pull request closed without merging**